frontend/src/components/UserSearch.vue

```diff
@@ -187,12 +221,25 @@ export default {
     },
     searchUser: function(queryParameters) {
       this.userSearchLoadingStatus = true;
+      let roleId = this.roleInput.trim();
 
       UsersRepository.get(
         "?briefRepresentation=false&first=0&max=300" + queryParameters
       )
         .then(response => {
-          this.searchResults = response.data;
+          let results = response.data;
+          if (this.advancedSearchSelected && roleId !== "") {
+            // REVIEW - Should this search be cached somwhow?
```

Comment:
I'd rather not until performance becomes a problem. I don't want to have to deal with cache invalidation too early.